File: MyApp2/app/src/main/java/com/android/killqwerty/myapp/myapp2/MainActivity.java
package com.android.killqwerty.myapp.myapp2;

import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.Toast;


public class MainActivity extends AppCompatActivity {
    Button button;
    Toast toast;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        setButtons();
    }
    public void setButtons(){
        button = findViewById(R.id.button2);
        button.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                if(toast != null){
                    toast.cancel();
                }
                toast = Toast.makeText(getApplicationContext(), R.string.pivko,
                        Toast.LENGTH_LONG);
                toast.show();
            }
        });
    }
}

File: RetrofitTraining/app/src/main/java/com/android/killqwerty/myapp/retrofittraining/MainActivity.kt
package com.android.killqwerty.myapp.retrofittraining

import android.app.Activity
import android.os.Bundle
import android.support.v7.widget.LinearLayoutManager
import android.support.v7.widget.RecyclerView
import android.util.Log
import android.widget.Button
import retrofit2.Call
import retrofit2.Callback
import retrofit2.Response

class MainActivity : Activity() {
    private lateinit var recyclerView: RecyclerView
    private lateinit var viewAdapter: RecyclerView.Adapter<*>
    private lateinit var viewManager: RecyclerView.LayoutManager
    private var myListOfPosts: List<User> = emptyList()
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.main)
        findViewById<Button>(R.id.my_btn_users).setOnClickListener { conectingToUsers() }
    }
    fun conectingToUsers(){
        if (myListOfPosts.isEmpty()) {
            val postsApi = IApiService.create()
            val response = postsApi.getAllUsers()
            response.enqueue(object : Callback<List<User>> {
                override fun onResponse(call: Call<List<User>>, response: Response<List<User>>) {
                    myListOfPosts = response.body()!!
                    createView()
                    Log.d("callback", "onResponse: ${response.body()!!}")
                }

                override fun onFailure(call: Call<List<User>>, t: Throwable) {
                    Log.d("callBack", "onFailure")
                }
            })
        }else {
            Log.d("callback","it is else and not response")
            createView()
        }
    }
    fun createView(){
        viewManager = LinearLayoutManager(this)
        viewAdapter = MyAdapter(myListOfPosts)
        recyclerView = findViewById<RecyclerView>(R.id.recycler_view).apply {
            setHasFixedSize(true)
            layoutManager = viewManager
            adapter = viewAdapter
        }
    }

//    override fun onSaveInstanceState(outState: Bundle?) {
//        outState.run {
//        }
//        super.onSaveInstanceState(outState)
//    }
//
//    override fun onRestoreInstanceState(savedInstanceState: Bundle?) {             такс, потом будет сохранение чтобы не делать запрос, пока не понял как проще то
//        savedInstanceState.getParcelableArrayList<User>()
//        super.onRestoreInstanceState(savedInstanceState)
//    }

    override fun onDestroy() {
        super.onDestroy()
    }
}
